src/uoapi/course/course_info.py

```diff
@@ -1,87 +1,128 @@
 import sys
-from time import sleep, perf_counter as pf
-import urllib
-import json
-import itertools as it
 
-import requests
 from bs4 import BeautifulSoup
-#import pandas as pd
+from typing import (
+    Callable,
+    List,
+    TypeVar,
+    cast
+)
+
+import requests
 import regex as re
 
 from uoapi.course import patterns as pt
 from uoapi.course import Prereq
 
-#requests.packages.urllib3.util.ssl_.DEFAULT_CIPHERS += ':DES-CBC3-SHA'
-#timetable_url = 'https://web30.uottawa.ca/v3/SITS/timetable/Search.aspx'
+try:
+    from re import (
+        Match
+    )
+except ImportError:
+    # https://github.com/python/cpython/blob/4bb332cfd1f9740b1e31d2d8b8bf1bedca3439ff/Lib/re.py#L272
+    import sre_compile as _sre
+    Match = type(_sre.compile('', 0).match(''))
+
+# requests.packages.urllib3.util.ssl_.DEFAULT_CIPHERS += ':DES-CBC3-SHA'
+# timetable_url = 'https://web30.uottawa.ca/v3/SITS/timetable/Search.aspx'
 
 # Course Info Parameters
 course_url = 'https://catalogue.uottawa.ca/en/courses/'
 
+T = TypeVar('T')
+
 #############################################################################
 # COURSE INFO SCRAPING
 #############################################################################
 
-def _extract_codes(string, return_all = True):
+
+def _extract_codes(string: str, return_all: bool = True):
```

Comment:
I assume this is the return type? (I don't 100% recall...).
Also, it seems like `return_all` isn't used here (and indeed it is set to `False` when this function is called), so I would say we can safely remove it (and propagate the signature change).
```suggestion
def _extract_codes(string: str) -> List[str]:
```